addons/sale_project_stock/models/project_project.py

```diff
@@ -0,0 +1,18 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class ProjectProject(models.Model):
+    _inherit = 'project.project'
+
+    def _get_picking_action(self, action_name, picking_type=None):
+        result = super()._get_picking_action(action_name, picking_type)
+
+        if property_warehouse := self.env.user.property_warehouse_id:
```

Comment:
Actually, it is better to first check if `picking_type` is not `None` otherwise we will fetch `property_warehouse_id` for nothing if `picking_type is None`. 🙂 

```suggestion
        if picking_type and (property_warehouse := self.env.user.property_warehouse_id):
```

I am not sure the parenthesis are needed

cc @deneuvillem

Reply:
Nice catch :) I think we can keep the parenthesis for the sake of readability

Reply:
We need parenthesis(high precedence) as walrus operator is assignment operator so comparison and assignment can't be done in a single expression in Python(If we need to keep current order in if statement).

We dont need parenthesis if we do assignment first(walrus) and comparsion
```python
if property_warehouse := self.env.user.property_warehouse_id and picking_type:
```
The above mentioned solution(xbo's) seems to be perfect

Reply:
okay thanks for the confirmation